src/exporters/kafka/exporter.rs

```diff
@@ -0,0 +1,594 @@
+        let key = if !config.partition_logs_by_resource_attributes {
+            // Default: empty message key
+            String::new()
+        } else {
+            // When partition_logs_by_resource_attributes is enabled, use hash of resource attributes as key
+            // Expect logs to contain only one ResourceLogs after splitting
+            let hash = extract_resource_attributes_hash_from_logs(logs);
+            hash.map_or(String::new(), hex::encode)
+        };
+        let payload = builder.build_message(logs)?;
+        Ok((key, payload))
+    }
+
+    fn telemetry_type() -> &'static str {
+        "logs"
+    }
+
+    /// Split logs by resource attributes when partition_logs_by_resource_attributes is enabled
+    /// Each ResourceLogs becomes its own message, matching the Go implementation
+    fn split_for_partitioning(config: &KafkaExporterConfig, data: Vec<Self>) -> Vec<Vec<Self>> {
+        if config.partition_logs_by_resource_attributes {
+            // Split each ResourceLogs into separate groups, one per ResourceLogs
+            // This matches the Go implementation where each resourceLogs becomes a separate message
+            data.into_iter().map(|log| vec![log]).collect()
+        } else {
+            vec![data]
+        }
+    }
+}
+
+/// Generic Kafka exporter for OpenTelemetry data
+pub struct KafkaExporter<Resource>
+where
+    Resource: KafkaExportable,
+{
+    config: KafkaExporterConfig,
+    producer: FutureProducer,
+    request_builder: KafkaRequestBuilder<Resource, Resource::Request>,
+    rx: BoundedReceiver<Vec<Resource>>,
+    topic: String,
+    encoding_futures: FuturesUnordered<EncodingFuture>,
+    send_futures: FuturesUnordered<SendFuture>,
+}
+
+impl<Resource> Debug for KafkaExporter<Resource>
+where
+    Resource: KafkaExportable,
+{
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("KafkaExporter")
+            .field("config", &self.config)
+            .field("topic", &self.topic)
+            .field("telemetry_type", &Resource::telemetry_type())
+            .finish_non_exhaustive()
+    }
+}
+
+impl<Resource> KafkaExporter<Resource>
+where
+    Resource: KafkaExportable,
+{
+    /// Create a new Kafka exporter
+    pub fn new(
+        config: KafkaExporterConfig,
+        rx: BoundedReceiver<Vec<Resource>>,
+        topic: String,
+    ) -> Result<Self> {
+        // Build Kafka producer
+        let client_config = config.build_client_config();
+        let producer: FutureProducer = client_config.create().map_err(|e| {
+            KafkaExportError::ConfigurationError(format!("Failed to create producer: {}", e))
+        })?;
+
+        let request_builder = Resource::create_request_builder(config.serialization_format.clone());
+
+        Ok(Self {
+            config,
+            producer,
+            request_builder,
+            rx,
+            topic,
+            encoding_futures: FuturesUnordered::new(),
```

Comment:
Do we need to use FuturesOrdered here to maintain logical ordering of messages for the output topic? I checked what we have today, it looks like the OTLP exporter is using Unordered but the http-based ones are Ordered, so we have some inconsistency.

Reply:
Done